register/views.py

```diff
@@ -1,8 +1,17 @@
-from registration.backends.default.views import ActivationView
-from registration.backends.default.views import RegistrationView
+from django.views.generic import TemplateView, RedirectView
+from django.shortcuts import reverse
 
-from django.views.generic import TemplateView
+from registration.backends.default.views import RegistrationView  # noqa
 
 
 class ProfileView(TemplateView):
     template_name = 'register/profile.html'
+
+
+class NextRedirectView(RedirectView):
+    def get_redirect_url(self, *args, **kwargs):
+        try:
+            redirect = self.request.session.pop('next')
```

Comment:
I've wondered wether to pop it out or not. I'm still unsure about this decision. It may be worth a discussion to check the cases.